### DISCUSSION TITLE
Selecting kernel on web vscode using ipad

### DISCUSSION THREAD

**Discussion by zhy21345** [microsoft/vscode-jupyter#15998] [Questions and Answers] [answered]
Hi, im using a web vscode(TIL) and tried to do some jupyter stuff. I did installed the jupyter and python extensions but it doesn’t allow me to select other kernels than using an existing jupyter server. Any hints or did i mess up anything?

edited: i tried like connecting to codespace but it seems that terminal is not supported and therefore can’t do anything with it.

**Comment by DonJayamanne** [comment]
Thank you for filing this issue and sorry you are running into this.
You seem to have a few things going on
Lets focus on one thing at a time.

> s but it doesn’t allow me to select other kernels than using an existing jupyter server. Any hints or did i mess up anything?

Does this issue repro when using the browser on your desktop?
Pelase can you share the screenshots of this working in the desktop browser and not working in ipad.
This way i know exactly what you're referring to.

**Reply by zhy21345** [reply]
Sure, heres the screenshot:
![IMG_1678](https://github.com/user-attachments/assets/5fafbe39-d791-40bf-8f5f-dccbcd4030a6)

This is the first time i do something like this on ipad and web vscode. I don’t have the desktop with me but it’s pretty straight forward before: i have the python environment deployed in my desktop, and i install the jupyter extension on vscode, and the first time it runs it will find my python environment and ask which python i want to use as kernel. I am not sure about how to do this in web vscode, some tutorials suggest that i will need a remote machine as well.  Is there anything i need to set up first before running jupyter on this setting?

**Reply by zhy21345** [reply]
btw the codespace connection option seems to be relying on terminal which is not an option on web vscode idk

**Reply by DonJayamanne** [reply, marked as answer]
> This is the first time i do something like this on ipad and web vscode.

No worries, i've provided a link with details.

>  am sure about how to do this in web vscode, some tutorials suggest that i will need a remote machine as well. Is there anything i need to set up first before running jupyter on this setting?

In the browser there's no python environment. Python is something you need to install on a physical device.
You can install on your machine or you can use code spaces & then use the browser to connect to them.
See here for more details https://docs.github.com/en/codespaces/developing-in-a-codespace/getting-started-with-github-codespaces-for-machine-learning
